### ISSUE TITLE
bug: amazons3:putobjectlockconfigurationaync is throwing exception

### ISSUE THREAD

**Issue by krishgopi007** [localstack/localstack#9670]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

 I am using AWS sdk along with localstack to validate s3permissions. . 

The amazons3config is provided with following parameters while connecting to localstack  :
ServiceUrl="http://localhost:4566/",
ForcePathStyle=true,
AuthenticationRegion="us-east-1"

While using localstack and debugging through the code I can see that 
amazons3:putobjectlockconfigurationasync is throwing an exception "bucket needs to be empty".

Observed this exception in community version as well as pro trial version.

This behaviour is not observed while using AWS instead of localstack.

Is this a bug. Is there a solution for this issue.

### Expected Behavior

amazons3:putobjectlockconfigurationasync should not throw this exception 

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

Localstack is run using docker compose file

### Environment

```markdown
- OS:
- LocalStack:
```

**Comment by bentsku** [comment]
Could you please share exactly the step your are taking while setting this configuration?

**Comment by bentsku** [comment]
I would need to know what AWS SDK version you are using, as from what I can see, we are not raising an exception with this specific message: `bucket needs to be empty`, which is often raised when trying to delete a bucket. 

If you could provide us with a small reproducible sample, that would be great.

In general, `PutObjectLockConfiguration` is an operation that can only be done on buckets created with `ObjectLockEnabledForBucket` set to `true`. Could this be the issue?

**Comment by krishgopi007** [comment]
I got this error in AWScli also. These are the steps.
1. Created a bucket with object lock enabled. 
2. Added an object to the bucket. (  .txt file ) 
3. Putobjectlock configuration with {"objectenabled": "Enabled"}

**Comment by bentsku** [comment]
I can confirm it work with LocalStack `3.0.0`:
```bash
$ awslocal s3api create-bucket --bucket test --object-lock-enabled-for-bucket
{
    "Location": "/test"
}
$ awslocal s3api put-object --bucket test --key test.txt --body test.txt 
{
    "ETag": "\"b1946ac92492d2347c6235b4d2611184\"",
    "ServerSideEncryption": "AES256",
    "VersionId": "dEpswJGkA2gQlu1Bfh8lcg"
}
$  awslocal s3api put-object-lock-configuration \
    --bucket test \                      
    --object-lock-configuration '{ "ObjectLockEnabled": "Enabled", "Rule": { "DefaultRetention": { "Mode": "COMPLIANCE", "Days": 50 }}}'
# no error here, it just returns
$ awslocal s3api get-object-lock-configuration --bucket test
{
    "ObjectLockConfiguration": {
        "ObjectLockEnabled": "Enabled",
        "Rule": {
            "DefaultRetention": {
                "Mode": "COMPLIANCE",
                "Days": 50
            }
        }
    }
}
```
If you can confirm it works for you, I'll close the issue!

**Comment by krishgopi007** [comment]
Hi @bentsku, 
The issue is resolved by using localstack 3.0.0. Thankyou so much for your response.

Regards,
Krishna G

**Comment by bentsku** [comment]
Awesome, I'm glad to hear! `latest` should also contain the new S3 implementation, you can get the real `latest` version by running `docker pull localstack/localstack:latest`. 

I'll close the issue, thanks again for the feedback!
